kouyama/resNavi/script.js

```diff
@@ -0,0 +1,204 @@
+const gnaviUrl = `https://api.gnavi.co.jp/RestSearchAPI/v3/`; // ぐるなびAPIのURL
+const keyid = 'e2d1f212bb0ddfdd774c23e22bedf31e'; // ぐるなびAPIのkeyid
+let data; // URLとして送信するオブジェクト
+let $search_name; // 店名検索に入力された文字列
+let $free_word; // フリーワード検索に入力された文字列
+let $thc; // 検索ヒット数totalhitcount
+let $thcPage; // 総検索ヒットページ数
+let page = 1; // 10ページごとの移動用
+
+
+// 検索ボタンをクリックされた時
+$('#search').on('click', function() {
+  $('#thc_result').html('<img src="loading.gif">'); // ローディング画像を表示
+  page = 1; // ページ番号初期化
+  data = { keyid : keyid }; // 前回の検索dataを初期化
+
+  $search_name = $('#search_name').val(); // 「店名」検索フォームに入力された文字列
+  $free_word = $('#free_word').val(); // 「フリーワード」検索フォームに入力された文字列
+  let $range = $('#range').val(); // 検索する範囲を指定する変数
+
+  if (!$search_name && !$free_word && !$range) { // 店名・フリーワード・範囲どれも入力されていない場合
```

Comment:
検索ワードが無いときに`alert()`が呼ばれる処理ですが、`$range`が意図しないものになっているように見えます。
検索ワードも範囲も指定しない状態で検索ボタンを押すと、`$range`には"0"（文字列）が入ります。
"0"はtrue扱いのものなので、否定`!`をつけるとfalseになります。

つまり、条件式がこんな感じになります。
```js
// 検索ワードも範囲も指定しない状態で検索ボタンを押した場合
if (true && true && false) {
  alert('検索ワードを入力してください');
  return;
}
```

単純に"0"かどうかをみるのはどうでしょうか。
```suggestion
  if (!$search_name && !$free_word && $range === "0") { // 店名・フリーワード・範囲どれも入力されていない場合
```